src/common/sckaddr.cpp

```diff
@@ -180,7 +180,11 @@ namespace
 #if defined(HAVE_GETHOSTBYNAME) && \
     !defined(HAVE_FUNC_GETHOSTBYNAME_R_6) && \
     !defined(HAVE_FUNC_GETHOSTBYNAME_R_5) && \
-    !defined(HAVE_FUNC_GETHOSTBYNAME_R_3)
+    !defined(HAVE_FUNC_GETHOSTBYNAME_R_3) || \
+   ( defined(HAVE_GETHOSTBYADDR) && \
+    !defined(HAVE_FUNC_GETHOSTBYADDR_R_6) && \
+    !defined(HAVE_FUNC_GETHOSTBYADDR_R_5) && \
+    !defined(HAVE_FUNC_GETHOSTBYADDR_R_3) )
```

Comment:
@vadz:
The Android NDK defines HAVE_FUNC_GETHOSTBYNAME_R_6, but does not define HAVE_GETHOSTBYNAME.  The sckaddr.cpp code defines HAVE_GETHOSTBYNAME.  As a result, since HAVE_FUNC_GETHOSTBYNAME_R_6 is defined, the function deepCopyHostent() is not included, and the build fails.
A simple solution is to undefine HAVE_FUNC_GETHOSTBYNAME_R_6 on Android.

Reply:
It looks like the best and simplest solution would be to just remove the test for `__ANDROID__` entirely as it seems that it doesn't need any special treatment any more. If we do this, and if `HAVE_FUNC_GETHOSTBYNAME_R_6` is already defined by configure, then `HAVE_FUNC_GETHOSTBYADDR_R_6` should be also defined, and `deepCopyHostent()` shouldn't be necessary at all. Could you please check if this works for you:
```diff
diff --git a/src/common/sckaddr.cpp b/src/common/sckaddr.cpp
index 3e252f0522..095e2481d6 100644
--- a/src/common/sckaddr.cpp
+++ b/src/common/sckaddr.cpp
@@ -106,11 +106,11 @@
     #endif
 #endif // __WINDOWS__

-#ifdef __ANDROID__
-    #ifndef HAVE_GETHOSTBYNAME
-    #define HAVE_GETHOSTBYNAME
-    #endif
-#else // these functions are missing on android (see netdb.h)
+// we assume that we have gethostbyaddr_r() if and only if we have
+// gethostbyname_r() and that it uses the similar conventions to it (see
+// comment in configure)
+#define HAVE_GETHOSTBYADDR HAVE_GETHOSTBYNAME
+
 #ifdef HAVE_FUNC_GETHOSTBYNAME_R_3
     #define HAVE_FUNC_GETHOSTBYADDR_R_3
 #endif
@@ -120,12 +120,6 @@
 #ifdef HAVE_FUNC_GETHOSTBYNAME_R_6
     #define HAVE_FUNC_GETHOSTBYADDR_R_6
 #endif
-#endif // __ANDROID__/!__ANDROID__
-
-// we assume that we have gethostbyaddr_r() if and only if we have
-// gethostbyname_r() and that it uses the similar conventions to it (see
-// comment in configure)
-#define HAVE_GETHOSTBYADDR HAVE_GETHOSTBYNAME

 // the _r functions need the extra buffer parameter but unfortunately its type
 // differs between different systems and for the systems which use opaque
```
?

BTW, how do you build for Android currently? It would be great to set up a GitHub CI build using Android to test it as currently it's not being tested at all anywhere. If you could contribute a PR doing this, it would be very welcome!

Reply:
@vadz 
Thanks for the comments.
As is typical, it seems that on some older Android versions (at least version 5.x), the libraries have gethostbyname_r(), but they DO NOT have gethostbyaddr_r().  Or at least not such that the configure script can find it.  So the assumption in the code "we assume that we have gethostbyaddr_r() if and only if we have gethostbyname_r()" fails.
More modern version of Android (e.g. Android 9) contain both functions, so all is well with your patch.
For widest Android version compatibility, it is probably best to assume neither function is available, and so resort to the deepCopyHostent() runtime path.  This will require a conditional Android compilation, as before.
I build OpenCPN for Android on a local machine.  We have a plan to get this into our general OpenCPN CI process, but we are going slow pending the wx3.2 release patches under discussion here.  I will take a look at the wxWidgets CI process, and see if I can come up with a simple Android app build for CI.  I agree that this would be a good thing, both for wx code quality, as well as instructional for potential wxQt/Android app builders.

Reply:
I need to complain now. This will break our build, and our commitment to support at least down to Android 5.  We have a large user base down there....

Reply:
Oh, sorry, I didn't realize you needed this yourself.

OK, let's fix this properly then. But looking at this code, I can't help wondering if we still need this (decades old) mess. Could we use `getaddrinfo()` everywhere instead? I.e. is it available in all Android versions you support?

Reply:
@vadz...
Investigating.  As it turns out, my code does not actually ever call gethostbyaddr_r().  That function is only needed to satisfy the Qt libraries which are dlopen-ed at run-time.
So, the simple solution now is to commit your patch above, and I will work around the missing function by including a NOP stub in my native code.  That way, the runtime linkage will be satisfied, and all will be well.
I have tested this approach on Android5, and it works fine.
So I suggest the simplest, non-breaking, solution now.

Reply:
OK, thanks, I'll leave it like this for now, but please let us know if you run into problems with the current master and also please check if `get{addr,name}info()` are available under Android if you do. TIA!

Reply:
@vadz
All good now with master.
Also, get{addr,name}info() seem to be both available in the Android libraries.  I have not tested at runtime, though.  Will do so and confirm.
Thanks
Dave